### PR TITLE
Fix/oo 88 adding zusatzkoerbe when changing vertrieb in mainabo

### PR DESCRIPTION
**Pull request by mcmpp**
Integrating the fix in v2.2 too

### CODE REVIEW

File: src/main/scala/ch/openolitor/stammdaten/KorbHandler.scala [hunk 1]
```diff
@@ -308,22 +308,51 @@ trait KorbHandler extends KorbStatusHandler
...
+      /* create or delete basket for the main abo*/
+      val offenLieferungenForMainAbo = stammdatenWriteRepository.getLieferungenOffenByAbotyp(abo.abotypId)
+      offenLieferungenForMainAbo map { lieferung =>
         if (lieferung.vertriebId == originalAbo.vertriebId) {
           deleteKorb(lieferung, originalAbo)
         }
         if (lieferung.vertriebId == abo.vertriebId) {
           upsertKorb(lieferung, abo, abotyp)
         }
-        recalculateNumbersLieferung(lieferung)
```
Comment (AlwinEgger) [inline src/main/scala/ch/openolitor/stammdaten/KorbHandler.scala:325]: recalculateNumbersLieferung() has been removed, but is included in the part for the Zusatzabo. Have you done this by purpose?

**Review by AlwinEgger — APPROVED** [review]
Very difficult to tell... @fabianpage or @rawyler lease do a second review.

File: src/main/scala/ch/openolitor/stammdaten/KorbHandler.scala (hunk 1 above)
Comment (fabianpage) [reply src/main/scala/ch/openolitor/stammdaten/KorbHandler.scala:325]: recalculateNumbersLieferung for all Main Abo's is done at Line 355.

**Comment by fabianpage** [comment]
Very hard to review because so much is happening in this code.

**Pull request merged**